diff --git a/camera/provider/2.4/default/Android.bp b/camera/provider/2.4/default/Android.bp
--- a/camera/provider/2.4/default/Android.bp
+++ b/camera/provider/2.4/default/Android.bp
@@ -215,7 +215,7 @@ cc_binary {
     proprietary: true,
     relative_install_path: "hw",
     srcs: ["external-service.cpp"],
-    compile_multilib: "32",
+    compile_multilib: "64",
     init_rc: ["android.hardware.camera.provider@2.4-external-service.rc"],
     shared_libs: [
         "android.hardware.camera.common@1.0",
